vllm/v1/engine/detokenizer.py

```diff
@@ -177,7 +193,8 @@ def __init__(self, tokenizer: PreTrainedTokenizerFast,
             skip_special_tokens=self.skip_special_tokens)
 
         self.tokenizer: Tokenizer = tokenizer._tokenizer
-
+        if self.reasoning_parser:
+            self.reasoning_parser = self.reasoning_parser(self.tokenizer)
```

Comment:
I think it would be clearer to add tokenizer to the superclass constructor and do this in the `BaseIncrementalDetokenizer` constructor.